lib/adapter.js

```diff
@@ -53,82 +54,59 @@ Adapter.prototype.connect = function (callback) {
         const tableName = self.params.database.migrationTable;
         this.db = new Pensuer.Db(self.params.database.name, settings);
         this.db.table(self.params.tables);
-        this.db.connect( (err) => {
-
-            if (err) {
-                return callback(err);
-            }
+        this.db.connect().then(() => {
 
-            this.db.establish({ [tableName]: { purge: false } }, (err) => {
-
-                if (err) {
-                    return callback(err);
-                }
+            return this.db.establish({ [tableName]: { purge: false } });
+        })
+            .then(() => {
 
                 return callback(null, { db: self.db });
-            });
-        });
+            })
+            .catch((err) => callback(err));
     });
 };
 
 Adapter.prototype.disconnect = function (callback) {
 
-    this.db.close(() => {
-
-        return callback();
-    });
+    this.db.close().then(callback);
```

Comment:
This does change the semantics a little bit - previously we would have swallowed the `close()` error (if that's even possible), whereas now we won't. Unsure if the earlier behavior was intentional?

Reply:
My intention was not to change the semantics, hence only adding the success handler in `then`, but I now realise that would leave an unhandled rejection, so I will update it to have a no-op